lib/server/middleware/maintenance.js

```diff
@@ -0,0 +1,35 @@
+import boolean from 'boolean';
+import SettingModel from '../models/setting';
+// import PageModel from '../models/page';
+import parseSettings from 'helpers/parse-settings';
+import ReactMaintenancePage from '../../shared/screens/public/screens/maintenance/index.js';
+import React from 'react';
+import {renderToString} from 'react-dom/server';
+
+export default async (req, res, next) => {
+  const settingsArr = await SettingModel
```

Comment:
there's currently no role based system property attached to the user

Plus: 
checking if the user is admin before knowing if maintenance mode is on would mean we would only know if we are in maintenance mode if the user is an admin. 

So we would have to get the settings either way and then do checks to see if user is admin